pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1860,6 +1880,87 @@ func handleDelete(ctx *domainContext, key string, status *types.DomainStatus) {
 		status.UUIDandVersion, status.DisplayName)
 }
 
+// DomainCreate is a wrapper for domain creation thru xlCreate or rktRun
+func DomainCreate(status *types.DomainStatus) (int, error) {
+
+	var domainID int
+	var err error
+
+	filename := xenCfgFilename(status.AppNum)
+	log.Infof("DomainCreate %s ... xenCfgFilename - %s\n", status.DomainName, filename)
+	if status.IsContainer {
+		// Use rkt tool
+		log.Infof("Using rkt tool ... ContainerImageID - %s\n", status.ContainerImageID)
+		domainID, err = rktRun(status, status.ContainerImageID, filename)
+	} else {
+		// Use xl tool
+		log.Infof("Using xl tool ... xenCfgFilename - %s\n", filename)
+		domainID, err = xlCreate(status.DomainName, filename)
+	}
+
+	return domainID, err
+}
+
+//  Launch app/container thru rkt
+func rktRun(status *types.DomainStatus, ContainerImageID string, xenCfgFilename string) (int, error) {
+
+	// STAGE1_XL_OPTS=-p STAGE1_SEED_XL_CFG=xenCfgFilename rkt --dir=<RKT_DATA_DIR> --insecure-options=image run <SHA> --stage1-path=/usr/sbin/stage1-xen.aci --uuid-file-save=uuid_file
+	log.Infof("rktRun %s - ContainerImageID %s\n", status.DomainName, ContainerImageID)
+	cmd := "rkt"
+	args := []string{
+		"--dir=" + persistRktDataDir,
+		"--insecure-options=image",
+		"run",
+		ContainerImageID,
+		"--stage1-path=/usr/sbin/stage1-xen.aci",
+		"--uuid-file-save=/persist/rkt/uuid_file",
+	}
+	stage1XlOpts := "STAGE1_XL_OPTS=-p"
+	stage1XlCfg := "STAGE1_SEED_XL_CFG=" + xenCfgFilename
+	log.Infof("Calling command %s %v\n", cmd, args)
+	log.Infof("Also setting env vars %s %s\n", stage1XlOpts, stage1XlCfg)
+	cmdLine := exec.Command(cmd, args...)
+	cmdLine.Env = append(os.Environ(), stage1XlOpts, stage1XlCfg)
+	stdoutStderr, err := cmdLine.CombinedOutput()
+	if err != nil {
+		log.Errorln("rkt run failed ", err)
+		log.Errorln("rkt run output ", string(stdoutStderr))
+		return 0, fmt.Errorf("rkt run failed: %s\n",
+			string(stdoutStderr))
+	}
+	log.Infof("rkt run done\n")
+
+	// Get Pod UUID
+	uuidData, err := ioutil.ReadFile("/persist/rkt/uuid_file")
+	if err != nil {
+		log.Errorf("Open /persist/rkt/uuid_file failed : %s\n", err)
+		return 0, fmt.Errorf("open /persist/rkt/uuid_file failed: %s\n", err)
+	}
+	status.PodUUID = strings.TrimSpace(string(uuidData))
```

Comment:
Instead of having this function that status as an arg and modifying it, I think to makdes sense to return the PodUUID to the caller and have it or its caller modify status.

Reply:
I will make the necessary changes.